### ISSUE TITLE
Questions based on Saved Questions is not using the same query for date filters leading to wrong results

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#15352]
**Describe the bug**
When using date filters on questions, which are used as base (Saved Questions) for other questions, then the filters does not include the same dates.

<details><summary>Possible workaround for Between filter is to add the time - click for example</summary>

**NOTE** It will still miss a minute from the end date

Base question:
![image](https://user-images.githubusercontent.com/1447303/112630637-2bd9bc00-8e36-11eb-811e-fbb44184b018.png)

Nested question:
![image](https://user-images.githubusercontent.com/1447303/112630830-722f1b00-8e36-11eb-9924-c23319b3f8ea.png)

</details>

----

**Example 1: Between**
1. Custom question > Sample Dataset > Orders
2. Filter "Created At" **Between** `02/01/2020` and `02/29/2020`, summarize Count, click Visualize
![image](https://user-images.githubusercontent.com/1447303/112629263-5aef2e00-8e34-11eb-9980-483129ac4363.png)
3. Result is **543** - save as "Q1"
The query generated:
```
SELECT count(*) AS "count"
FROM "PUBLIC"."ORDERS"
WHERE ("PUBLIC"."ORDERS"."CREATED_AT" >= timestamp with time zone '2020-02-01 00:00:00.000+01:00'
   AND "PUBLIC"."ORDERS"."CREATED_AT" < timestamp with time zone '2020-03-01 00:00:00.000+01:00')
```
4. Simple question > Saved Questions > "Q1" - the result is **519** instead of 543
The query generated - it's excluding the entire day of February 29th
```
SELECT "source"."count" AS "count"
FROM (SELECT count(*) AS "count" FROM "PUBLIC"."ORDERS"
WHERE "PUBLIC"."ORDERS"."CREATED_AT" BETWEEN timestamp with time zone '2020-02-01 00:00:00.000+01:00'
   AND timestamp with time zone '2020-02-29 00:00:00.000+01:00') "source"
LIMIT 1048576
```

----

**Example 2: After/Before**
1. Custom question > Sample Dataset > Orders
2. Filter "Created At" **After** `01/31/2020` and **Before** `03/01/2020`, summarize Count, click Visualize
![image](https://user-images.githubusercontent.com/1447303/112629156-38f5ab80-8e34-11eb-87f0-3a39481c4e1c.png)
3. Result is **543** - save as "Q2"
The query generated:
```
SELECT count(*) AS "count"
FROM "PUBLIC"."ORDERS"
WHERE ("PUBLIC"."ORDERS"."CREATED_AT" >= timestamp with time zone '2020-02-01 00:00:00.000+01:00'
   AND "PUBLIC"."ORDERS"."CREATED_AT" < timestamp with time zone '2020-03-01 00:00:00.000+01:00')
```
4. Simple question > Saved Questions > "Q2" - the result is **560** instead of 543
The query generated - the After filter is including the entire day of January 31st
```
SELECT "source"."count" AS "count"
FROM (SELECT count(*) AS "count" FROM "PUBLIC"."ORDERS"
WHERE ("PUBLIC"."ORDERS"."CREATED_AT" > timestamp with time zone '2020-01-31 00:00:00.000+01:00'
   AND "PUBLIC"."ORDERS"."CREATED_AT" < timestamp with time zone '2020-03-01 00:00:00.000+01:00')) "source"
LIMIT 1048576
```

----

**Example 3: On**
1. Custom question > Sample Dataset > Orders
2. Filter "Created At" **On** `02/01/2020`, summarize Count, click Visualize
![image](https://user-images.githubusercontent.com/1447303/112629881-25971000-8e35-11eb-8c9f-625f3465cc19.png)
3. Result is **17** - save as "Q3"
The query generated:
```
SELECT count(*) AS "count"
FROM "PUBLIC"."ORDERS"
WHERE ("PUBLIC"."ORDERS"."CREATED_AT" >= timestamp with time zone '2020-02-01 00:00:00.000+01:00'
   AND "PUBLIC"."ORDERS"."CREATED_AT" < timestamp with time zone '2020-02-02 00:00:00.000+01:00')
```
4. Simple question > Saved Questions > "Q3" - the result is **0** instead of 17
The query generated - the On filter will only find results, which was created at midnight of February 1st
```
SELECT "source"."count" AS "count"
FROM (SELECT count(*) AS "count" FROM "PUBLIC"."ORDERS"
WHERE "PUBLIC"."ORDERS"."CREATED_AT" = timestamp with time zone '2020-02-01 00:00:00.000+01:00') "source"
LIMIT 1048576
```

**Information about your Metabase Installation:**
Tested 0.35.4 thru 0.38.2 and `master`

**Additional context**
Similar to #12872, which was fixed in 0.37.4

**Comment by camsaul** [comment]
Fixed by #15555

**Comment by nemanjaglumac** [comment]
This issue had a `.Reproduced` label but the related repro wasn't skipped.
